src/Libraries/Microsoft.Extensions.AI/ChatCompletion/FunctionInvokingChatClient.cs

```diff
@@ -840,6 +863,17 @@ FunctionResultContent CreateFunctionResultContent(FunctionInvocationResult resul
         return result;
     }
 
+    /// <summary>This method will invoke the function within the try block.</summary>
+    /// <param name="context">The function invocation context.</param>
+    /// <param name="cancellationToken">Cancellation token.</param>
+    /// <returns>The function result.</returns>
+    protected virtual async Task<object?> InvokeFunctionAsync(FunctionInvocationContext context, CancellationToken cancellationToken)
```

Comment:
While we're making changes here, probably good to change this to return ValueTask.

Reply:
I thought about that and wasn't sure what would be the impact of `Task` for `ValueTask` implementers.